# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/RemoteTextureMap.h
# Commit: b9f43d8a6c2d
# Full Hash: b9f43d8a6c2d9504dde533d3db13111ae4d31d06
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-07-12 21:56:41
# Regressor Bug: 1777426
# File Overlap Count: 2
# Description:
#   Bug 1777426 - Support fast readbacks in CopyToSwapChain with async present. r=sotaro,jgilbert
#   
#   Currently CopyToSwapChain creates spurious copies of the back buffer when SharedSurfaces aren't exportable (= ToSurfaceDescriptor returns Nothing from SharedSurface_Basic). These then later get read back into a CPU memory buffer when PresentFrontBufferToCompositor is used to send the buffer to RemoteTextureMap. This has associated performance and memory costs.
#   
#   Conceptually, we want Present/CopyToSwapChain to just do the right thing and automatically push buffers to RemoteTextureMap, rather than secondarily needing a hidden call to PresentFrontBufferToCompositor. Then we can get rid of the need to create front buffers whose only purpose is to shuttle data to PresentFrontBufferToCompositor which then shuttles it RemoteTextureMap.
# ==============================================================================

diff -r 0a598b1533e6 -r b9f43d8a6c2d gfx/layers/RemoteTextureMap.h
--- a/gfx/layers/RemoteTextureMap.h	Tue Jul 12 06:49:11 2022 +0000
+++ b/gfx/layers/RemoteTextureMap.h	Tue Jul 12 06:56:19 2022 +0000
@@ -51,10 +51,10 @@
   void RegisterTextureOwner(const RemoteTextureOwnerId aOwnerId);
   void UnregisterTextureOwner(const RemoteTextureOwnerId aOwnerId);
   void UnregisterAllTextureOwners();
-  void PushTexure(const RemoteTextureId aTextureId,
-                  const RemoteTextureOwnerId aOwnerId,
-                  UniquePtr<TextureData>&& aTextureData,
-                  const std::shared_ptr<gl::SharedSurface>& aSharedSurface);
+  void PushTexture(const RemoteTextureId aTextureId,
+                   const RemoteTextureOwnerId aOwnerId,
+                   UniquePtr<TextureData>&& aTextureData,
+                   const std::shared_ptr<gl::SharedSurface>& aSharedSurface);
   UniquePtr<TextureData> CreateOrRecycleBufferTextureData(
       const RemoteTextureOwnerId aOwnerId, gfx::IntSize aSize,
       gfx::SurfaceFormat aFormat);
@@ -146,11 +146,11 @@
   // gl::SharedSurface is pushed only when the surface needs to be kept alive
   // during TextureHost usage. The texture data and the surface might be
   // recycled when TextureHost is destroyed.
-  void PushTexure(const RemoteTextureId aTextureId,
-                  const RemoteTextureOwnerId aOwnerId,
-                  const base::ProcessId aForPid,
-                  UniquePtr<TextureData>&& aTextureData,
-                  const std::shared_ptr<gl::SharedSurface>& aSharedSurface);
+  void PushTexture(const RemoteTextureId aTextureId,
+                   const RemoteTextureOwnerId aOwnerId,
+                   const base::ProcessId aForPid,
+                   UniquePtr<TextureData>&& aTextureData,
+                   const std::shared_ptr<gl::SharedSurface>& aSharedSurface);
   void RegisterTextureOwner(const RemoteTextureOwnerId aOwnerId,
                             const base::ProcessId aForPid);
   void UnregisterTextureOwner(const RemoteTextureOwnerId aOwnerIds,
